src/invoice.rs

```diff
@@ -208,6 +244,33 @@ impl FromStr for RgbInvoice {
                 }
             };
 
+        let mut query: IndexMap<String, String> = IndexMap::new();
+        if let Some(q) = uri.query() {
+            if let Ok(query_str) = q.decode().into_string() {
```

Comment:
If `q.decode()` does URI decoding, than it should be applied after `split('&')`; otherwise we may end up with a string with multiple ampersants, which comes from `%`-encoded representation.

Reply:
improved the implementation. switched to decoding query param tokens separately. not using fluent-uri's examples directly as those silently skip query parameters that don't successfully decode, while we want to report the error to the user

added a test for %-encoded query params. IMO it has to require the invoice string to be encoded back in order to pass, so I did that via the urlencoding crate, which has been added to the dependencies. please check this is fine with you